invenio_rdm_records/resources/serializers/signposting/schema.py

```diff
@@ -0,0 +1,215 @@
+    type = fields.Method(serialize="serialize_type")
+
+    def serialize_anchor(self, obj, **kwargs):
+        """Seralize to landing page URL."""
+        return record_url_for(_app="ui", pid_value=obj["id"])
+
+    def serialize_author(self, obj, **kwargs):
+        """Serialize author(s).
+
+        For now, the first linkable identifier is taken.
+        """
+
+        def pick_linkable_id(identifiers):
+            for id_dict in identifiers:
+                try:
+                    url = idutils.to_url(
+                        id_dict["identifier"], id_dict["scheme"], url_scheme="https"
+                    )
+                    if url:
+                        return url
+                    else:
+                        continue
+                except Exception as e:
+                    print("pick linkable id exception")
+                    print(e)
+            return None
+
+        metadata = obj["metadata"]
+        result = [
+            {"href": pick_linkable_id(c["person_or_org"].get("identifiers", []))}
+            for c in metadata.get("creators", [])
+        ]
+        result = [r for r in result if r["href"]]
+        return result or missing
+
+    def serialize_cite_as(self, obj, **kwargs):
+        """Serialize cite-as."""
+        identifier = obj.get("pids", {}).get("doi", {}).get("identifier")
+        if not identifier:
+            return missing
+
+        try:
+            url = idutils.to_url(identifier, "doi", "https")
+        except Exception as e:
+            url = None
+
+        return [{"href": url}] if url else missing
+
+    def serialize_describedby(self, obj, **kwargs):
+        """Serialize describedby."""
+        # Has to be placed here to prevent circular dependency.
+        from invenio_rdm_records.resources.config import record_serializers
+
+        result = [
+            {"href": record_url_for(_app="api", pid_value=obj["id"]), "type": mimetype}
+            for mimetype in sorted(record_serializers)
+        ]
+
+        return result or missing
+
+    def serialize_item(self, obj, **kwargs):
+        """Serialize item."""
+        file_entries = obj.get("files", {}).get("entries", [])
+
+        result = [
+            {
+                "href": download_url_for(pid_value=obj["id"], filename=entry["key"]),
+                "type": entry["mimetype"],
+            }
+            for entry in file_entries
+        ]
+
+        return result or missing
+
+    def serialize_license(self, obj, **kwargs):
+        """Serialize license.
+
+        Note that we provide an entry for each license (rather than just 1).
+        """
+        rights = obj["metadata"].get("rights", [])
+
+        result = [{"href": right["link"]} for right in rights if right.get("link")]
+
+        return result or missing
+
+    def serialize_type(self, obj, **kwargs):
+        """Serialize type."""
+        resource_type = obj["metadata"]["resource_type"]
+
+        rt_record = vocabulary_service.read(
```

Comment:
I think this is going to hit the database, so it should be using some sort of cache to not get a performance hit. @ntarocco was looking into similar issues with the search results.